group-coordinator/src/main/java/org/apache/kafka/coordinator/group/generic/GenericGroup.java

```diff
@@ -738,6 +742,84 @@ public String generateMemberId(String clientId, Optional<String> groupInstanceId
             .orElseGet(() -> clientId + MEMBER_ID_DELIMITER + UUID.randomUUID());
     }
 
+    /**
+     * Validates that (1) the group instance id exists and is mapped to the member id
+     * if the group instance id is provided; and (2) the member id exists in the group.
+     *
+     * @param memberId          The member id.
+     * @param groupInstanceId   The group instance id.
+     * @param operation         The operation.
+     *
+     * @throws UnknownMemberIdException
+     * @throws FencedInstanceIdException
+     */
+    public void validateMember(
```

Comment:
do we want this to be public?

Reply:
Yes because it is used in the GroupMetadataManager as well.